pkg/cloudprovider/metrics/cloudprovider_test.go

```diff
@@ -0,0 +1,55 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/aws/karpenter-core/pkg/cloudprovider"
+	. "github.com/onsi/gomega"
+)
+
+func TestGetErrorTypeLabelValue(t *testing.T) {
```

Comment:
Old school table tests. How do you feel about ginkgo? It's what we use across the project, and supports things like test focusing e.g. `FOCUS="my-test-name make e2etest`

Reply:
Happy to re-use ginkgo. @jonathan-innis how do we feel about evolving the current `fake.CloudProvider` stuff to inject error responses. Do we want a new fake type w/ method responses or is there an established pattern in the code to get those error responses back using the existing fake pattern(s)?

Reply:
I could go either way. Updating the `fake.CloudProvider` with `NextError` on each function sounds reasonable to me. You could also create a custom fake for this test suite and that also works.

Reply:
I don't see any existing test coverage that includes prometheus validation, should be do a follow-up PR that establishes the patterns for all metrics testing in the codebase?

Reply:
There's a semi-established pattern of doing metrics checking [here](https://github.com/aws/karpenter-core/blob/main/pkg/controllers/metrics/pod/suite_test.go#L59), but I agree, we can take this in a follow-up. I created this issue for tracking: https://github.com/aws/karpenter-core/issues/327